### PR TITLE
refactor: Prepare code for installing older Northstar versions

### PR DESCRIPTION
**Pull request by GeckoEidechse**
This PR aims to refactor Northstar installation code to make it easier to later add functionality to install a specific Northstar versions

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
@catornot you wanna review this PR? At least all the changes in here are backend/Rust stuff and AFAIK you have the most Rust knowledge + FlightCore knowledge combo out of the people I know ^^

File: src-tauri/src/lib.rs [hunk 1]
```diff
@@ -279,31 +279,24 @@ async fn do_install(
...
-
     let index = thermite::api::get_package_index().unwrap().to_vec();
     let nmod = index
         .iter()
         .find(|f| f.name.to_lowercase() == northstar_package_name.to_lowercase())
         .ok_or_else(|| panic!("Couldn't find Northstar on thunderstore???"))
         .unwrap();
 
+    // Use passed version or latest if no version was passed
+    let version = version_number.unwrap_or_else(|| nmod.latest.clone());
```
Comment (catornot) [inline src-tauri/src/lib.rs:293]: this could be a borrow instead to avoid cloning and you borrow this anyway later

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -327,9 +327,23 @@ async fn install_northstar_caller(
     window: tauri::Window,
     game_path: String,
     northstar_package_name: Option<String>,
+    version_number: Option<String>,
 ) -> Result<bool, String> {
     log::info!("Running");
-    match install_northstar(window, &game_path, northstar_package_name).await {
+
+    // Get Northstar package name (`Northstar` vs `NorthstarReleaseCandidate`)
+    let northstar_package_name = match northstar_package_name {
```
Comment (catornot) [inline src-tauri/src/main.rs:335]: the match case can be replaced with `map` and `unwrap_or` ;)

**Review by catornot — CHANGES_REQUESTED** [review]
ok

File: src-tauri/src/lib.rs (hunk 1 above)
Comment (GeckoEidechse) [reply src-tauri/src/lib.rs]: So like this a769530b59c0a46813f1ac881dda628267438329 ? ^^"

File: src-tauri/src/main.rs (hunk 2 above)
Comment (GeckoEidechse) [reply src-tauri/src/main.rs:335]: So like

```rust
    let northstar_package_name = northstar_package_name
        .map(|name| {
            if name.len() <= 1 {
                "Northstar".to_string()
            } else {
                name
            }
        })
        .unwrap_or("Northstar".to_string());
```
? ^^"

Comment (catornot) [reply src-tauri/src/main.rs:335]: yeah

File: src-tauri/src/lib.rs (hunk 1 above)
Comment (catornot) [reply src-tauri/src/lib.rs]: yes

File: src-tauri/src/main.rs (hunk 2 above)
Comment (GeckoEidechse) [reply src-tauri/src/main.rs]: Done in 05483e672b95c664abb1a4df1918006c3b956ebb

**Review by catornot — APPROVED** [review]
good

**Pull request merged**